### DISCUSSION TITLE
Any way how to bypass such question " Enter passphrase for key /tmp/semaphore/access_key_XXXXXXXX"

### DISCUSSION THREAD

**Discussion by i-kit** [semaphoreui/semaphore#872] [Q&A]
When I tried to use pass protected ssh key got next question in "cli" on playbook run:
"Enter passphrase for key /tmp/semaphore/access_key_XXXXXXXX"
At same way key without protections works well.

**Comment by fiftin** [comment]
Currently no way

**Reply by jdhayes** [reply]
In theory one could use `ssh-agent`, no?

**Reply by i-kit** [reply]
Bulshit btw if not designed from from scratch - seems this app has huge security violation due to this and can't be used in prod ... Only for pet-projects :(

**Reply by jdhayes** [reply]
I believe it is possible to run an `ssh-agent` in the background and by setting the correct environment variables one could (in theory) use a passphrase protected key. I only mention this as to facilitate a conversation which could bring about a new feature or solution for you.

Additionally, many opensource projects start out as pet projects. You can always fork your own copy and add/fix whatever you want (I did).
If you want something with enterprise support, why not try Red Hat's Ansible Tower?

**Reply by Johnybe** [reply]
Hi,

Great job for Semaphore it's so sexy , we need this for ssh encypted key

https://docs.ansible.com/ansible-tower/2.4.0/html/userguide/credentials.html

**_Private Key Passphrase: If the SSH Private Key used is protected by a password, you can configure a Key Password for the private key. This password may be stored encrypted in the Tower database, if entered. Alternatively, you can configure Tower to ask the user for the password as necessary by selecting “Ask at runtime?”. In these cases, a dialog opens when the job is launched, prompting the user to enter the password and password confirmation._**

**Reply by jeansieb** [reply]
I guess the following line in the documentation is then incorrect about the key store, as there is no native support for password protected keys:
"At this time, semaphore does support using SSH keys that are password protected."

**Comment by george-pago-ro** [comment]
Any updates on this? Or does someone have an idea how to use a ssh key with passphase?
I tried setting the ansible_ssh_private_key_file in the inventory, add the key to ssh-agent but i still get "Permission denied (publickey,password)" even if running ssh is working.

**Comment by Akshay-Hegde** [comment]
Any solution in 2026 like awx/tower prompt  for private key passphrase and prompt for password at runtime ? 

Or any alternate solution to manage passphrase, password and vault password at runtime instead of storing permanently in database
